docs/src/model_simulation/examples/activation_time_distribution_measurement.md

```diff
@@ -0,0 +1,58 @@
+```@example activation_time_distribution_measurement
+using Catalyst
+sa_model = @reaction_network begin
+    @parameters Kᵢ Kₐ K=Kᵢ
+    @discrete_events [0.0] => [K ~ Kₐ]
+    v0 + hill(X,v,K,n), 0 --> X
+    deg, X --> 0
+end
+```
+Next, to perform stochastic simulations of the system we will create an `SDEProblem`. Here, we will need to assign parameter values to $Kᵢ$ and $Kₐ$, but not to $K$ (as its value is controlled by its default and the event). Also note that we start the simulation at a time $t = -200 < 0$. This ensures that by the input time ($t = 0$), the system has (more or less) reached its (inactive) steady state distribution. It also means that the activation time can be measured exactly as the simulation time at the time of activation (as this will be the time from the input at $t = 0$).
+```@example activation_time_distribution_measurement
+u0 = [:X => 10.0]
+tspan = (-200.0, 2000.0)
+ps = [:v0 => 0.1, :v => 2.5, :Kᵢ => 1000.0, :Kₐ => 40.0, :n => 3.0, :deg => 0.01]
+sprob = SDEProblem(sa_model, u0, tspan, ps)
+nothing # hide 
+```
+We can now create a simple `EnsembleProblem` and perform an ensemble simulation (as described [here](@ref ensemble_simulations)). Please note that the system s an event which modifies its parameters, hence we must add the `safetycopy = true` argument to `EnsembleProblem` (else, subsequent simulations would start with $K = Kₐ$).
+```@example activation_time_distribution_measurement
+using Plots, StochasticDiffEq
+eprob = EnsembleProblem(sprob; safetycopy = true)
+esol = solve(eprob, ImplicitEM(); trajectories = 10)
+plot(esol)
+```
+Here we see how, after the input time, the system (randomly) switches from the inactive state to the active one (several examples of this, bistability-based, activation have been studied in the literature, both in models and experiments[^1][^2]).
+
+Next, we wish to measure the distribution of these activation times. First we will create a [callback](https://docs.sciml.ai/DiffEqDocs/stable/features/callback_functions/) which terminates the simulation once it has reached a threshold. This both ensures that we do not have to expend unnecessary computer time on the simulation after its activation, and also enables us to measure the activation time as the final time point of the simulation. Here we will use a [discrete callback](https://docs.sciml.ai/DiffEqDocs/stable/features/callback_functions/#SciMLBase.DiscreteCallback). By looking at the previous simulations, we determine $X = 100$ as a suitable activation threshold. We use the `terminate!` function to terminate the simulation once this value has been reached.
+```@example activation_time_distribution_measurement
+condition(u, t, integrator) = integrator[:X] > 100.0
+affect!(integrator) = terminate!(integrator)
+callback = DiscreteCallback(condition, affect!)
```

Comment:
Can you not use a functional affect and store this in the system?

Reply:
Does system events store non-trivial affects like `termiante`? If it is easier to do I am happy to change, however, it doesn't hurt to show how more complicated events can be implemented through callbacks.